tasks/managed/push-rpm-to-koji/push-rpm-to-koji.yaml

```
@@ -233,10 +240,16 @@ spec:
               echo "ERROR: No Koji build target found in the container image annotations."
               exit 1
           fi
-          # For draft builds, remove "-candidate" tag suffix if present and
-          # append "-draft".
-          if [[ "$draft" == "true" && "$koji_target" != *-draft ]]; then
-              koji_tag=${koji_target%-candidate}-draft
+
+          # For draft builds, keep any sidetag as is, otherwise remove
+          # "-candidate" tag suffix if present and append "-draft".
+          if [[ "$draft" == "true" ]]; then
+            is_sidetag=$(is_sidetag "$koji_target")
+            if [[ "$is_sidetag" == "true" ]]; then
+                koji_tag=${koji_target}
+            elif [[ "$koji_target" != *-draft ]]; then
+                koji_tag=${koji_target%-candidate}-draft
+            fi
           else
```

correct me if I'm wrong, but it looks like there's a missing else clause 

```suggestion
          if [[ "$draft" == "true" ]]; then
            is_sidetag=$(is_sidetag "$koji_target")
            if [[ "$is_sidetag" == "true" ]]; then
                koji_tag=${koji_target}
            elif [[ "$koji_target" != *-draft ]]; then
                koji_tag=${koji_target%-candidate}-draft
            else
                koji_tag=${koji_target}
            fi
          else
```

---

Ah, you are right, thanks. I'm thinking about reverting it to the previous version, without the nested ifs.

---

Fixed.

---

@elenagerman please check the updated version. If you're happy with it now, this PR can be merged.

---

@elenagerman ping